src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -0,0 +1,94 @@
+        }
+
+        /// <summary>
+        /// Runs the api compat for the tuples in the queue.
+        /// </summary>
+        public void RunApiCompat()
+        {
+            foreach (var apicompatTuples in _queue.Distinct())
+            {
+                // TODO: Add Assembly version check.
+                // TODO: Add optimisations tuples.
+                // TODO: Run it Asynchronously.
+                using (Stream leftAssemblyStream = GetFileStreamFromPackage(apicompatTuples.leftAssemblyPackagePath, apicompatTuples.leftAssemblyRelativePath))
+                using (Stream rightAssemblyStream = GetFileStreamFromPackage(apicompatTuples.rightAssemblyPackagePath, apicompatTuples.rightAssemblyRelativePath))
+                {
+                    IAssemblySymbol leftSymbols = new AssemblySymbolLoader().LoadAssembly(apicompatTuples.assemblyName, leftAssemblyStream);
+                    IAssemblySymbol rightSymbols = new AssemblySymbolLoader().LoadAssembly(apicompatTuples.assemblyName, rightAssemblyStream);
+                    ApiComparer differ = new();
+
+                    IEnumerable<CompatDifference> differences = differ.GetDifferences(leftSymbols, rightSymbols).Where(t => !_checker.Contain(t.DiagnosticId, t.ReferenceId));
+
+                    if (differences.Any())
+                    {
+                        _log.LogError(apicompatTuples.compatibilityReason);
+                        _log.LogError(apicompatTuples.header);
+                    }
+
+                    foreach (CompatDifference difference in differences)
+                    {
+                        _log.LogError(difference.ToString());
+                    }
+                }
+            }
+            _queue.Clear();
+        }
+
+        /// <summary>
+        /// Queues the api compat for 2 assemblies.
+        /// </summary>
+        /// <param name="leftPackagePath">Path to package containing left assembly.</param>
+        /// <param name="leftRelativePath">Relative left assembly path in package.</param>
+        /// <param name="rightPackagePath">Path to package containing right assembly.</param>
+        /// <param name="rightRelativePath">Relative right assembly path in package.</param>
+        /// <param name="assemblyName">The name of the assembly.</param>
+        /// <param name="compatibiltyReason">The reason for assembly compatibilty.</param>
+        /// <param name="header">The header for the api compat diagnostics.</param>
+        public void QueueApiCompat(string leftPackagePath, string leftRelativePath, string rightPackagePath, string rightRelativePath, string assemblyName, string compatibiltyReason, string header)
+        {
+            _queue.Add((leftPackagePath, leftRelativePath, rightPackagePath, rightRelativePath, assemblyName, compatibiltyReason, header));
+        }
+
+        private static Stream GetFileStreamFromPackage(string packagePath, string entry)
+        {
+            MemoryStream ms = new MemoryStream();
```

Comment:
MemoryStream is disposable and should be wrapped inside a using.

Reply:
https://github.com/dotnet/sdk/pull/16992/files/5946d6d65fd62f77f3114f87bdcd193a9a0ef9c2#diff-e367e9c9b80d9fae09d950fca7e12c7506297162a5a9d1ee2a6cf0214c11ee8dR40 

we are using this "using statement" here